plancontext/platform.go

```diff
@@ -5,14 +5,8 @@ import (
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-var (
-	// Default platform.
-	// FIXME: This should be auto detected using buildkit
-	defaultPlatform = specs.Platform{
-		OS:           "linux",
-		Architecture: "amd64",
-	}
-)
+// Default platform.
+var defaultPlatform = platforms.Normalize(platforms.DefaultSpec())
```

Comment:
[`DefaultSpec`](https://github.com/containerd/containerd/blob/e079e4a155c86f07bbd602fe6753ecacc78198c2/platforms/defaults_darwin.go#L29) isn't the good function since it takes `OS` from `runtime.GOOS`, instead you should call [`Default`](https://github.com/containerd/containerd/blob/e079e4a155c86f07bbd602fe6753ecacc78198c2/platforms/defaults_darwin.go#L39) because it sets the os by default to `linux`

Reply:
Interesting... `Default` doesn't return a platform but a `MatchComparer`. Trying to see where buildkit uses that to compare platforms as it's not used in the `state.go` file (https://github.com/moby/buildkit/blob/master/client/llb/state.go#L33) when setting the constraints.